### PR TITLE
Wa 1(adding theme swither)

### PR DESCRIPTION
**Pull request by uselesswhale1**
Added comp 'theme-switcher' as sub-comp of 'header'. Did markup and scss styles for it

### CODE REVIEW

File: weather-app/src/app/header/theme-switcher/theme-switcher.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div id="header__theme-switcher">
+  <label id="" class="theme-switcher__label">
+    <input type="checkbox" [value]="value" (change)="onToggle(value)" id="" />
+    <span class="switcher--slider round"></span>
```
Comment (Anurov-Prihodko) [inline weather-app/src/app/header/theme-switcher/theme-switcher.component.html:4]: why do you use here two "--"?

File: weather-app/src/app/header/header.component.html [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
+          class="logo__icon"
+          viewBox="0 0 75 65"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <rect
+            width="75"
+            height="65"
+            rx="5"
+            transform="matrix(1 0 0 -1 0 65)"
```
Comment (Anurov-Prihodko) [inline weather-app/src/app/header/header.component.html:15]: need to remove extra space

File: weather-app/src/app/app.module.ts [hunk 3]
```diff
@@ -5,9 +5,15 @@ import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
 import { FooterComponent } from './footer/footer.component';
 import { HeaderComponent } from './header/header.component';
+import { ThemeSwitcherComponent } from './header/theme-switcher/theme-switcher.component';
```
Comment (Anurov-Prihodko) [inline weather-app/src/app/app.module.ts:8]: maybe swither must be inside a header component?

File: weather-app/src/app/app.component.html [hunk 4]
```diff
@@ -1,4 +1,3 @@
-<main class="main-container">
-  <router-outlet></router-outlet>
-</main>
+<app-header></app-header>
+<!--<div class="main-container"></div>-->
```
Comment (Anurov-Prihodko) [inline weather-app/src/app/app.component.html:2]: why you delete main and routing?

File: weather-app/src/app/app.module.ts [hunk 5]
```diff
@@ -4,23 +4,12 @@ import { BrowserModule } from '@angular/platform-browser';
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
 import { FooterComponent } from './footer/footer.component';
-import { HomeComponent } from './home/home.component';
-import { DetailsComponent } from './details/details.component';
-import { ForecastComponent } from './forecast/forecast.component';
+import { HeaderComponent } from './header/header.component';
```
Comment (Anurov-Prihodko) [inline weather-app/src/app/app.module.ts:7]: why you delete imports of the main components?

File: weather-app/src/app/app.module.ts [hunk 6]
```diff
@@ -4,23 +4,12 @@ import { BrowserModule } from '@angular/platform-browser';
...
-    FooterComponent,
-    HomeComponent,
-    DetailsComponent,
-    ForecastComponent
-  ],
-  imports: [
-    BrowserModule,
-    AppRoutingModule
-  ],
+  declarations: [AppComponent, HeaderComponent, FooterComponent],
```
Comment (Anurov-Prihodko) [inline weather-app/src/app/app.module.ts]: and here you delete declarations of the main components?

File: weather-app/src/app/header/theme-switcher/theme-switcher.component.html [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<div id="header__theme-switcher">
+  <label id="" class="theme-switcher__label">
+    <input type="checkbox" [value]="value" (change)="onToggle(value)" id="" />
```
Comment (dmitry-chirva) [inline weather-app/src/app/header/theme-switcher/theme-switcher.component.html:3]: remove id

File: weather-app/src/app/header/theme-switcher/theme-switcher.component.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-theme-switcher',
+  templateUrl: './theme-switcher.component.html',
+  styleUrls: ['./theme-switcher.component.scss'],
+})
+export class ThemeSwitcherComponent implements OnInit {
+  constructor() {}
+  lightTheme = 'light';
```
Comment (dmitry-chirva) [inline weather-app/src/app/header/theme-switcher/theme-switcher.component.ts:10]: add readonly

File: weather-app/src/app/header/theme-switcher/theme-switcher.component.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  darkTheme = 'dark';
+
+  value = this.lightTheme;
```
Comment (dmitry-chirva) [inline weather-app/src/app/header/theme-switcher/theme-switcher.component.ts:13]: change value to theme

File: weather-app/src/app/header/theme-switcher/theme-switcher.component.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  ngOnInit(): void {}
+
+  onToggle(value: string): void {
+    console.log(value);
```
Comment (dmitry-chirva) [inline weather-app/src/app/header/theme-switcher/theme-switcher.component.ts:17]: 1) remove console
2) this.theme = value === this.lightTheme ? this.darkTheme:  this.lightTheme

**Pull request closed without merging**